wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/DevfileService.java

```diff
@@ -50,6 +52,9 @@
 @Path("/devfile")
 public class DevfileService extends Service {
 
+  private static final String LOCAL_FILES_REFERENCES_ARE_NOT_SUPPORTED =
+      "Devfile Service does not support tools which require local files content. You can use factory from Github URL instead.";
```

Comment:
Are we limiting ourselves to github here on purpose?

Reply:
Now, it's the only one git hosting service that we support for factories.